scripts/testTimezones.js

```diff
@@ -0,0 +1,118 @@
+﻿/* eslint-disable jest/no-commented-out-tests */
+/* eslint-disable no-use-before-define */
+
+const { spawn, execSync } = require('child_process');
+const os = require('os');
+
+/* eslint-disable-next-line import/no-extraneous-dependencies */
+const ianaWin = require('windows-iana');
+
+// This script runs date-related tests in multiple timezones to make sure that we don't have any
+// timezone-related heisenbugs. For example, this will fail if the system runtime is east of GMT:
+//
+// ```js
+// test('timezone heisenbug', () => {
+//   const d = new Date('2020-03-16');
+//   expect(getYYYYMMDD(d)).toBe('2020-03-16'); // sometimes we get 2020-03-15
+// });
+// ```
+
+// WARNING: On Windows, this script actually changes the system timezone. If the script doesn't exit
+// normally, you may be left in the wrong timezone. You can set it back by running this from the
+// command line:
```

Comment:
Would it be possible to add a process.on(`uncaughtException`) to reset the timezone in case we changed it?

Reply:
The more likely scenario for ending up with the wrong timezone is that the whole process gets interrupted (e.g. terminal is closed), and there's nothing we can do about that in code. I think @hyperknot is right, we can add a step to run these in CI, and a developer who is making code changes that touch timezones can run `yarn test:tz` as needed, and be aware that it might mess with their computer's time. As long as a normal `yarn test` doesn't run these I think it's fine.